test/ng/directive/inputSpec.js

```diff
@@ -2748,7 +2749,8 @@ describe('input', function() {
           expect($rootScope.value).toBe(30);
 
           // 0.3 - 0.2 === 0.09999999999999998
-          $rootScope.$apply('min = 0.2; step = 0.09999999999999998; value = 0.3');
+          $rootScope.$apply('min = 0.2; step = (0.3 - 0.2)');
+          helper.changeInputValueTo('0.3');
```

Comment:
Not changing the scope value (e.g. `$rootScope.$apply('value = '0.3')`), because `ngModel` does not overwrite programmatically changed scope values unless the validity of the input changes. Depending on the order in which the validators are run and the previous state, this may lead to the scope value remaining `0.3` instead of being set to undefined.